### ISSUE TITLE
Non-nullabe Int foreign key,  "defaultvalue : 0" when AddColumn.

### ISSUE THREAD

**Issue by raphael-theriault** [dotnet/efcore#30759]
Step to reproduce : 

- Set "Enable" into your project properties for "Nullable" option.
- Use Int as Primary Key
- Create table X with just a PK.
- Create table Y with a foreign key referencing X :  Non nullabe, no default value.  We will call this column, 'A'
- Add-migration & Update-database.
- Create a new foreign key in Y referencing X.  Non nullable, no default value; We will call this column, 'B'
- Add-migration & Update-database.

Column A will not have a default value set.
Column B will have a default value of 0 (zero)­.

In short.
MigrationBuilder, when using CreateTAble, will not add any default value parameter for a Int column.
However, when MigrationBuilder uses AddColumn. It will put add `defaultValue: 0`.

EF Core version: 7.0.2 
Target framework: .net 7

Addons : While looking at the whole consistency of my DB, I realized that not only int are affected by this bug but also string column.  Probably other types too...

**Comment by ajcvickers** [comment]
@raphael-theriault Can you please attach the project described above that reproduces this behavior so that we can investigate?

**Comment by raphael-theriault** [comment]
@ajcvickers , will put this on my to-do.  I would need to create a brand new project because this one is full of our client information.  In the meantime, if you start a new project on the side feel free to try it .

**Comment by raphael-theriault** [comment]
@ajcvickers , I just had a doubt in my mind, I might have posted this at the wrong place.
I am using Pomelo (MySql).  The reason why I posted here is because I supposed that the role of Pomelo is outside the creation of the migrations, where the problem is. 
In my comprehension EntityFramwork manage the translation of Code to Migration.
So if Pomelo has something to do when I used "add-migration". The problem might be related to Pomelo and thus, I would try it in SqlServer.
Thank you,

**Comment by ajcvickers** [comment]
**EF Team Triage:** Closing this issue as the requested additional details have not been provided and we have been unable to reproduce it.

*BTW this is a canned response and may have info or details that do not directly apply to this particular issue. While we'd like to spend the time to uniquely address every incoming issue, we get a lot traffic on the EF projects and that is not practical. To ensure we maximize the time we have to work on fixing bugs, implementing new features, etc. we use canned responses for common triage decisions.*
